aspnet/fundamentals/startup.rst

```diff
@@ -1,8 +1,196 @@
+
+.. note:: To see an example of how ``IApplicationBuilder`` is implemented, have a look at `Microsoft.AspNet.Http.ApplicationBuilder.cs <https://github.com/aspnet/HttpAbstractions/blob/eb0fe6a92ada31440e6db44d23b64f4d3d7240c0/src/Microsoft.AspNet.Http/ApplicationBuilder.cs>`_.
+
+
+The ConfigureServices method
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Your `Startup` class can optionally include a ``ConfigureServices`` method, which is called before ``Configure``. This is important, because some features like ASP.NET MVC require certain services to be added in ``ConfigureServices`` before they can be wired up to the request pipeline.
+
+Just as with ``Configure``, it is recommended that features that require substantial setup within ``ConfigureServices`` be wrapped up in extension methods on ``IApplicationBuilder``. You can see in this example from the default web site template that several ``Add[Something]`` extension methods are used to configure the app to use Entity Framework, Identity, and MVC:
+
+.. literalinclude:: ../../samples/WebApplication1/src/WebApplication1/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 48-87
+	:dedent: 8
+	:emphasize-lines: 8,14,34
+
+Adding services to the services container makes them available within your application via :doc:`dependency-injection`. Just as the ``Startup`` class is able to specify dependencies its methods require as parameters, rather than hard-coding to a specific implementation, so too can your controllers and other classes in your application.
+
+The ``ConfigureServices`` method is also where you should add configuration option classes, like ``AppSettings`` in the example above, that you would like to have available in your application (:doc:`learn more <configuration>`).
+
+Understanding request delegates
+-------------------------------
+
+ASP.NET 5 implements the Open Web Interface for .NET (OWIN). OWIN provides a standardized way for web applications to communicate with web servers, and specifies request delegates that respond to web requests. Request delegates are used to build the :ref:`request pipeline <request-pipeline>` that will be used to handle each incoming HTTP request to your application. Request delegates are configured using `Run, Map, and Use`_ extension methods on the ``IApplicationBuilder`` type that is passed into the ``Configure`` method in the ``Startup`` class. Request delegates, also known as :doc:`middleware`,  can be specified in-line as anonymous methods, or they can be defined in reusable classes.
```

Comment:
OWIN is it's own topic (#71). OWIN is not the primary design for request delegates or middleware.